warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py

```diff
@@ -248,7 +252,85 @@ def send_commands_from_testdata(self, testdatafile, **args):
                     finalresult = "ERROR"
                 finalresult = finalresult and result
             responses_dict[key] = response_dict
-        return finalresult, responses_dict
+            td_resp_dict = self.update_resp_ref_in_data_repo(details_dict, system_name, session_name,
+                                                             finalresult, responses_dict, td_resp_dict, resp_key_list)
+        return finalresult, td_resp_dict
+
+    def update_resp_ref_in_data_repo(self, details_dict, system_name, session_name,
+                                     finalresult, responses_dict, td_resp_dict, resp_key_list=[]):
+        """
+        -Updates the response reference key and value the respective session_id
+        -If the testcase and testdata file has two different system name, then it takes the td file
+        sys tag as priority and updates the response reference in the particular session_id
+
+            :Arguments:
+                1.details_dict = Contains all the details of the testdata block
+                2.system_name = Contains the system name given in the testcase
+                3.session_name = Contains the session name given in the testcase
+                4.responses_dict = Contains the response reference as a dictionary
+
+            :Returns:
+                td_resp_dict = dict
+        """
+        td_sys_list = []
+        td_session_list = []
+        td_session_id = ''
+
+        def get_sys_list_session_list(details_dict, td_sys_list, td_session_list):
+            """
+                Fetching the system name and session name from details_dict if available,
+                else takes from test case.
+            """
+
+            for i in details_dict["sys_list"]:
+                # If sys_list in None or if sys_tag in td file has only subsystem name, then it
+                # takes from the test case else fetches from the td file.
+                if i is None or i is '' or i.startswith('['):
+                    td_sys_list.append(system_name)
+                else:
+                    td_sys_list.append(i)
+
+            for k in details_dict["session_list"]:
+                # If session name not available it td file, it fetches from the test case.
+                if k is None or k is '':
+                    td_session_list.append(session_name)
+                else:
+                    td_session_list.append(k)
+            return td_sys_list, td_session_list
+
+        if finalresult:
+            if len(resp_key_list) != len(td_sys_list):
+                for j in resp_key_list:
+                    td_sys_list, td_session_list = \
+                     get_sys_list_session_list(details_dict, td_sys_list, td_session_list)
+            else:
+                td_sys_list, td_session_list = \
+                  get_sys_list_session_list(details_dict, td_sys_list, td_session_list)
+            for title_row, temp_resp_dict in responses_dict.iteritems():
```

Comment:
multiple response references in a single cmd should map to the same system, hence this logic should be revisited.